src/shmem_team.c

```diff
@@ -0,0 +1,455 @@
+    shmem_internal_bit_clear(psync_pool_avail, N_PSYNC_BYTES, SHMEMX_TEAM_WORLD_INDEX);
+    shmem_internal_bit_clear(psync_pool_avail, N_PSYNC_BYTES, SHMEMX_TEAM_SHARED_INDEX);
+
+    for (size_t i = 0; i < max_teams; i++) {
+        shmem_internal_team_pool[i] = NULL;
+    }
+    shmem_internal_team_pool[SHMEMX_TEAM_WORLD_INDEX] = &shmem_internal_team_world;
+    shmem_internal_team_pool[SHMEMX_TEAM_SHARED_INDEX] = &shmem_internal_team_shared;
+
+    return 0;
+}
+
+void shmem_internal_team_fini(void)
+{
+    /* Destroy all undestroyed teams */
+    for (long i = 0; i < shmem_internal_params.TEAMS_MAX; i++) {
+        if (shmem_internal_team_pool[i] != NULL)
+            shmem_internal_team_destroy(shmem_internal_team_pool[i]);
+    }
+
+    free(shmem_internal_team_pool);
+    shmem_internal_free(shmem_internal_psync_pool);
+    shmem_internal_free(psync_pool_avail);
+
+    return;
+}
+
+int shmem_internal_team_my_pe(shmem_internal_team_t *team)
+{
+    if (team == SHMEMX_TEAM_INVALID)
+        return -1;
+    else
+        return ((shmem_internal_team_t *)team)->my_pe;
+}
+
+int shmem_internal_team_n_pes(shmem_internal_team_t *team)
+{
+    if (team == SHMEMX_TEAM_INVALID)
+        return -1;
+    else
+        return ((shmem_internal_team_t *)team)->size;
+}
+
+void shmem_internal_team_get_config(shmem_internal_team_t *team, shmemx_team_config_t *config)
+{
+    shmem_internal_team_t *myteam = (shmem_internal_team_t *)team;
+    *config = myteam->config;
+    return;
+}
+
+int shmem_internal_team_translate_pe(shmem_internal_team_t *src_team, int src_pe,
+                                     shmem_internal_team_t *dest_team)
+{
+    int src_pe_world, dest_pe = -1;
+
+    if (src_team == SHMEMX_TEAM_INVALID || dest_team == SHMEMX_TEAM_INVALID)
+        return -1;
+
+    if (src_pe > src_team->size)
+        return -1;
+
+    src_pe_world = src_team->start + src_pe * src_team->stride;
+
+    if (src_pe_world < src_team->start || src_pe_world >= shmem_internal_num_pes)
+        return -1;
+
+    dest_pe = shmem_internal_pe_in_active_set(src_pe_world, dest_team->start, dest_team->stride,
+                                              dest_team->size);
+
+    return dest_pe;
+}
+
+int shmem_internal_team_split_strided(shmem_internal_team_t *parent_team, int PE_start, int PE_stride,
+                                      int PE_size, const shmemx_team_config_t *config, long config_mask,
+                                      shmem_internal_team_t **new_team)
+{
+
+    *new_team = SHMEMX_TEAM_INVALID;
+
+    if (parent_team == SHMEMX_TEAM_INVALID) {
+        return 0;
+    }
+
+    int global_PE_start = shmem_internal_team_pe(parent_team, PE_start);
+    int global_PE_end   = global_PE_start + PE_stride * (PE_size -1);
+
+    if (PE_size <= 0 || PE_stride < 1 || global_PE_start >= shmem_internal_num_pes ||
+        global_PE_end >= shmem_internal_num_pes) {
+        RAISE_WARN_STR("Invalid start, stride, or size in team_split operation");
```

Comment:
The error message prints the function name, filename, and line number.
```suggestion
        RAISE_WARN_MSG("Invalid team: start (%d), stride (%d), size (%d) -- first PE (%d), last PE (%d)\n",
                        PE_start, PE_stride, PE_size, global_PE_start, global_PE_end);
```

Reply:
Done in https://github.com/davidozog/sandia-shmem/pull/8 right?